WeatherMonitoringAndReportingService.BL/Program.cs

```diff
@@ -1,11 +1,99 @@
-﻿namespace WeatherMonitoringAndReportingService.BL
+﻿using WeatherMonitoringAndReportingService.BL.Parsers;
+using WeatherMonitoringAndReportingService.BL.Utilities;
+using WeatherMonitoringAndReportingService.BL.Enums;
+using WeatherMonitoringAndReportingService.BL.Weather;
+using WeatherMonitoringAndReportingService.BL.WeatherBots;
+
+namespace WeatherMonitoringAndReportingService.BL
 {
-    class Program
+    internal abstract class Program
     {
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            Console.WriteLine("Hello, there!");
-            Console.WriteLine("This is the real-time weather monitoring and reporting service application!");
+            var dataFormatChoice = ConsoleOutputUtility.GetDataFormat();
+
+            #region Declaring Instances
+
+            var jsonParser = new JsonWeatherDataParser();
+            var parserContext = new ParserContext(jsonParser);
+            var weatherPublisher = new WeatherBotPublisher();
+            weatherPublisher.SetConfigData(await ConfigDataUtility.ReadBotsConfigData());
+            var rainBot = new RainBot(weatherPublisher);
+            var snowBot = new SnowBot(weatherPublisher);
+            var sunBot = new SunBot(weatherPublisher);
```

Comment:
weatherPublisher is a good case for factory design pattern, and you can combine it with singleton pattern to provide one instance of  weatherPublisher that was configure and shipped (created) using factory